runtime/connectors/s3/s3.go

```diff
@@ -190,10 +195,121 @@ func (c connector) HasAnonymousAccess(ctx context.Context, env *connectors.Env,
 	if err != nil {
 		return false, fmt.Errorf("failed to open bucket %q, %w", conf.url.Host, err)
 	}
+	defer bucketObj.Close()
 
 	return bucketObj.IsAccessible(ctx)
 }
 
+func (c Connector) ListBuckets(ctx context.Context, env *connectors.Env) ([]string, error) {
+	creds, err := getCredentials(env)
+	if err != nil {
+		return nil, err
+	}
+	if creds == credentials.AnonymousCredentials {
+		return nil, fmt.Errorf("no credentials exist")
+	}
+
+	// Create a session that tries to infer the region from the environment
+	sess, err := session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable, // Tells to look for default region set with `aws configure`
+		Config: aws.Config{
+			Credentials: creds,
+		},
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	// no region found, default to us-east-1
+	if sess.Config.Region == nil || *sess.Config.Region == "" {
+		sess = sess.Copy(&aws.Config{Region: aws.String("us-east-1")})
+	}
```

Comment:
1. The region here is not used for the API. It is used to specify where we want to send our requests to (https://docs.aws.amazon.com/sdk-for-go/v1/developer-guide/configuring-sdk.html)
We can in fact send it to any location and the API should ideally succeed so that's why I just fallback to `us-east-1` if nothing is found in user configurations.
2. Good catch. This is a bug in the current resolution of region for a bucket as well. I will fix at both places.